storage/filestore/store.go

```diff
@@ -79,7 +109,45 @@ func (store *Store) Create(ctx context.Context, ref storage.BlobRef, size int64)
 	if err != nil {
 		return nil, Error.Wrap(err)
 	}
-	return newBlobWriter(ref, store, file), nil
+	return newBlobWriter(ref, store, storage.MaxStorageFormatVersionSupported, file), nil
+}
+
+// SpaceUsed adds up the space used in all namespaces for blob storage
+func (store *Store) SpaceUsed(ctx context.Context) (space int64, err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	var totalSpaceUsed int64
+	namespaces, err := store.GetAllNamespaces(ctx)
+	if err != nil {
+		return 0, Error.New("failed to enumerate namespaces: %v", err)
+	}
+	for _, namespace := range namespaces {
+		used, err := store.SpaceUsedInNamespace(ctx, namespace)
+		if err != nil {
+			return 0, Error.New("failed to sum space used: %v", err)
+		}
+		totalSpaceUsed += used
+	}
+	return totalSpaceUsed, nil
+}
+
+// SpaceUsedInNamespace adds up how much is used in the given namespace for blob storage
+func (store *Store) SpaceUsedInNamespace(ctx context.Context, namespace []byte) (int64, error) {
+	var totalUsed int64
+	err := store.ForAllKeysInNamespace(ctx, namespace, func(access storage.StoredBlobAccess) error {
+		statInfo, statErr := access.Stat(ctx)
+		if statErr != nil {
+			store.log.Sugar().Errorf("failed to stat: %v", statErr)
+			// keep iterating; we want a best effort total here.
+			return nil
```

Comment:
Errors in traversing the directory structure are relatively likely to happen in practice. We wouldn't want to stop iteration and get no results or incomplete results because of one permission error. We could accumulate all errors during iteration and hand them all back to the caller, but in all current cases the caller would still just log them and carry on.

If it becomes a need, I'll be fine with removing the log and passing accumulated errors back to the caller, but for now I think having a best-effort sum of space used is more useful.

Reply:
That sounds good to me and it's fine to remain as it's.